### PR TITLE
network events: add packet length and interface

### PR DESCRIPTION
**Pull request by roikol**
all network events have 'metadata' arg. add the packet length
and the originating interface to the 'metadata' arg.

## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

adding packet length and interface to trace.PktMeta arg.
this is the second PR regarding this issue.
once the PR #1762 will be merged, this PR should update the go.mod and it'll be ready to be merged as well.

Fixes: #1761

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -86,3 +86,5 @@ require (
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 )
+
+replace github.com/aquasecurity/tracee/types v0.0.0-20220412160215-ccc81e5e59cd => github.com/roikol/tracee/types v0.0.0-20220516100414-ca8aafbe6202
```
Comment (yanivagman) [inline go.mod:90]: Was this left here by mistake?

File: pkg/bufferdecoder/net_decoder.go [hunk 2]
```diff
@@ -51,21 +51,22 @@ func (decoder *EbpfDecoder) DecodeNetCaptureData(netCaptureData *NetCaptureData)
 	netCaptureData.PacketLength = binary.LittleEndian.Uint32(decoder.buffer[offset : offset+4])
 	netCaptureData.ConfigIfaceIndex = binary.LittleEndian.Uint32(decoder.buffer[offset+4 : offset+8])
 
-	decoder.cursor += int(netCaptureData.GetSizeBytes())
```
Comment (yanivagman) [inline pkg/bufferdecoder/net_decoder.go:54]: Can you please explain why did you remove this?

File: pkg/ebpf/net_proto.go [hunk 3]
```diff
@@ -14,16 +14,41 @@ import (
...
 }
 
+// resolveIfaceName replaces the interface number with interface name for 'metadata' arg
+func resolveIfaceName(evt *trace.Event, argName string, ifaceName string) error {
+	for idx, arg := range evt.Args {
+		if arg.Name == argName {
+			if metadata, isPktMeta := arg.Value.(trace.PktMeta); isPktMeta {
+				metadata.Iface = ifaceName
+				evt.Args[idx] = trace.Argument{
+					ArgMeta: arg.ArgMeta,
```
Comment (yanivagman) [inline pkg/ebpf/net_proto.go:41]: Don't we need to change the argument metadata in this case as well? (As the type is changed from int to string)

File: pkg/bufferdecoder/net_decoder.go (hunk 2 above)
Comment (roikol) [reply pkg/bufferdecoder/net_decoder.go:54]: so that when calling `DecodeNetPacketEvent`, the packetLength and IfaceIndex could be read again, and appended to the netadataArg.
the other option was to pass this as argument from `processNetEvents()` to `protocolProcessor()` and then to `callProtocolHandler()` and then to `netPacketProtocolHandler()`.

what do you think?

**Comment by roikol** [comment]
@yanivagman thanks for you review. i realised now i should do things differently here. i'll update the PR and then ask for second review.
thanks!

File: go.mod (hunk 1 above)
Comment (roikol) [reply go.mod]: nope, it was needed so the code will compile. since the PR to change the types (https://github.com/aquasecurity/tracee/pull/1762) was not merged when i opened this PR.
now the PR to change the types is merged, so no need for the `replace` anymore.

**Comment by rafaeldtinoco** [comment]
> @yanivagman thanks for you review. i realised now i should do things differently here. i'll update the PR and then ask for second review. thanks!

Moving this to DRAFT because of this comment. Please change it back once you're done and re-request review. Thanks!

**Review by yanivagman — APPROVED** [review]
LGTM
